SimG4Core/Application/python/g4SimHits_cfi.py

```diff
@@ -45,7 +45,7 @@
     DoFineCalo = cms.bool(False),
     # currently unused; left in place for future studies
     EminFineTrack = cms.double(10000.0),
-    FineCaloNames = cms.vstring('ECAL', 'HCAL', 'HGCal', 'HFNoseVol', 'VCAL'),
+    FineCaloNames = cms.vstring('ECAL', 'HCal', 'HGCal', 'HFNoseVol', 'VCAL'),
```

Comment:
@bsunanda , these are names of G4Regions. These names are used in many places. I would not change HCAL to HCal, because this will contradict to legacy and will require modifications in many places.

Reply:
No these are not the name of Regions. These are the names of top volume name in the geometry tree. I do not think this needs change in any other places

Reply:
@bsunanda , then implementation is not optimal, it should use G4Region, because it is one call to G4LogicalVolume, not more. If top volume name is used then computations are much more complicate.

Reply:
Implementation requires to check pointer to logicalvolume of a given level. This is a straightforward "==" between 2 unsigned integers. The level is also provided - I think this is quite a fast computation.

Reply:
but how you no the level in advance?

Reply:
This is provided as a parameter